### DISCUSSION TITLE
Lock orientation for filament UIView on iOS

### DISCUSSION THREAD

**Discussion by kpeeters** [google/filament#6348] [General]
This is a bit off-topic, but I'm hoping someone might know the answer, as none of the methods I have found so far work. 

I am looking to lock the orientation of the filament UIView on iOS, so that the system does not squash the render target into an area with the wrong aspect ratio after an orientation change. This seems to be the logic of the filament iOS demos, in which the orientation is fixed at Info.plist level (and rotation on orientation change is then handled by applying a rotation on the projection). However, I would like to allow all other views to be auto-rotated, and fix only the filament view. So I cannot fix the entire app to one orientation.

Is there a way to do this? I wrapped the filament UIView into a custom ViewController and did an override of `shouldAutoRotate` as well as `supportedInterfaceOrientations`, but that doesn't do anything; filament's UIView still auto-rotates.

**Comment by bejado** [comment]
I'm not sure iOS allows you to "fix" a single View to one orientation. I think instead you should update Filament's Viewport and Projection matrices after an orientation change. The gltf_viewer app takes this approach. See [`layoutSubviews` and `updateviewportAndCameraProjection`](https://github.com/google/filament/blob/main/ios/samples/gltf-viewer/gltf-viewer/FILModelView.mm).

**Comment by kpeeters** [comment]
What I ended up doing was wrapping my view in a separate `ViewController`, and then override its `willRotate(to toInterfaceOrientation: UIInterfaceOrientation, duration: TimeInterval)` in which I set the `transform` matrix of my view to counter-rotate. A single flag would have been nice, but this does the trick. Anything with `shouldAutoRotate` failed for me, and it's deprecated anyway. A search also comes up with many people who attempt a solution by manipulating `UIDevice`, but that really seems to be only for touching entire-device state (as the name suggests), not for individual views.

 If anyone wants details, drop me a line.
